kafka/conn.py

```diff
@@ -79,7 +79,7 @@ def _raise_connection_error(self):
 
     def _read_bytes(self, num_bytes):
         bytes_left = num_bytes
-        responses = []
+        result_buffer = six.BytesIO()
```

Comment:
Oh, right, we can totally do that. I thought we wanted to support elder pythons.

Now that I think of it, we might be able to use `memoryview` and such fast shenanigans since our floor version is 2.6. Let me get back to you on this.